omod/src/main/webapp/pages/results.gsp

```diff
@@ -32,21 +37,57 @@
                                           "bAutoWidth": false,
                                           "sDom": 'ft<"fg-toolbar ui-toolbar ui-corner-bl ui-corner-br ui-helper-clearfix datatables-info-and-pg "ip>'
                                       });
+
+        jq.fn.dataTableExt.ofnSearch['html'] = function (sData) {
+            return jq(sData).val();
+        };
+
+        jq('#rule_filter').change(function () {
+            resultTable.columns(0).search(jq(this).val()).draw();
+        });
+
+        jq("#patient_filter").autocomplete({
+            minLength: 0,
+            delay: 0,
+            source: Array.from(new Set(patients)),
+            select: function () {
+                resultTable.columns(3).search(jq(this).val()).draw();
+            }
+        });
+
     });
     <% } %>
 </script>
-<% if (rule == null) {
-%>
-<h2>${ui.message("dataintegrity.result.all.label")}</h2>
+
+<% if (rules == null) { %>
+    <h2>${ui.message("dataintegrity.result.all.label")}</h2>
+    <h3>${ui.message("dataintegrity.result.all.label")}</h3>
 <% } else { %>
-<h2>Results for ${rule.ruleName}</h2>
+    <h2>Results for ${rules.ruleName}</h2>
```

Comment:
@shivtej1505 Rules is now a list of rules not a single one so you need to correct that see previous comments